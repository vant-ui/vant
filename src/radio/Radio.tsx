import { defineComponent } from 'vue';
import { pick, createNamespace } from '../utils';
import { useParent } from '@vant/use';
import Checker, { checkerProps } from '../checkbox/Checker';
import { RADIO_KEY, RadioGroupProvide } from '../radio-group';

const [name, bem] = createNamespace('radio');

export default defineComponent({
  name,

  props: checkerProps,

  emits: ['update:modelValue'],

  setup(props, { emit, slots }) {
    const { parent } = useParent<RadioGroupProvide>(RADIO_KEY);

    const checked = () => {
      const value = parent ? parent.props.modelValue : props.modelValue;
      return value === props.name;
    };

    const toggle = () => {
      if (parent) {
        parent.updateValue(props.name);
      } else {
        emit('update:modelValue', props.name);
      }
    };

    return () => (
      <Checker
        v-slots={pick(slots, ['default', 'icon'])}
        bem={bem}
        role="radio"
        parent={parent}
        checked={checked()}
        onToggle={toggle}
        {...props}
      />
    );
  },
});
