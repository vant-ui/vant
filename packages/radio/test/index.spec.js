import Radio from '../';
import demoTest from '../../../test/demo-test';

demoTest(Radio);
