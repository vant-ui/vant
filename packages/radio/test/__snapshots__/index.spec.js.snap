// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders radio correctly 1`] = `
<section class="van-doc-demo-section demo-radio" style="background:undefined;">
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">基础用法</h2>
    <div class="demo-radio-group van-radio-group">
      <div class="van-radio"><span class="van-radio__input"><input type="radio" value="1" checked="checked" class="van-radio__control"> <i class="van-icon van-icon-checked" style="color:undefined;"> <!----></i></span> <span class="van-radio__label">单选框 1</span></div>
      <div class="van-radio"><span class="van-radio__input"><input type="radio" value="2" class="van-radio__control"> <i class="van-icon van-icon-check" style="color:undefined;"> <!----></i></span> <span class="van-radio__label">单选框 2</span></div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">禁用状态</h2>
    <div class="demo-radio-group van-radio-group">
      <div class="van-radio van-radio--disabled"><span class="van-radio__input"><input type="radio" disabled="disabled" value="1" class="van-radio__control"> <i class="van-icon van-icon-check" style="color:undefined;"> <!----></i></span> <span class="van-radio__label">单选框 1</span></div>
      <div class="van-radio van-radio--disabled"><span class="van-radio__input"><input type="radio" disabled="disabled" value="2" checked="checked" class="van-radio__control"> <i class="van-icon van-icon-checked" style="color:undefined;"> <!----></i></span> <span class="van-radio__label">单选框 2</span></div>
    </div>
  </section>
  <section class="van-doc-demo-block">
    <h2 class="van-doc-demo-block__title">与 Cell 组件一起使用</h2>
    <div class="van-radio-group">
      <div class="van-cell-group van-hairline--top-bottom">
        <div class="van-cell van-cell--clickable van-hairline">
          <!---->
          <div class="van-cell__title"><span>单选框1</span>
            <!---->
          </div>
          <div class="van-cell__value">
            <div class="van-radio"><span class="van-radio__input"><input type="radio" value="1" checked="checked" class="van-radio__control"> <i class="van-icon van-icon-checked" style="color:undefined;"> <!----></i></span>
              <!---->
            </div>
          </div>
          <!---->
        </div>
        <div class="van-cell van-cell--clickable van-hairline">
          <!---->
          <div class="van-cell__title"><span>单选框2</span>
            <!---->
          </div>
          <div class="van-cell__value">
            <div class="van-radio"><span class="van-radio__input"><input type="radio" value="2" class="van-radio__control"> <i class="van-icon van-icon-check" style="color:undefined;"> <!----></i></span>
              <!---->
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
  </section>
</section>
`;
